packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -181,7 +182,36 @@ export function createRootHooks<
                 },
               };
 
-              return client.query(...getClientArgs(queryKey, actualOpts));
+              const result = await client.query(
+                ...getClientArgs(queryKey, actualOpts),
+              );
+
+              if (isAsyncIterable(result)) {
+                const queryHash = hashKey(queryKey);
+
+                const queryCache = queryClient.getQueryCache();
+
+                const query = queryCache.build(queryClient, {
+                  queryKey,
+                  queryHash,
+                });
```

Comment:
i've never used the `.build()` method, but for all the methods I've used, I think I never needed both the `queryHash` *and* the `queryKey`. 

Does this still work with only the `queryHash`?
```suggestion
                const query = queryCache.build(queryClient, {
                  queryHash,
                });
```

Reply:
On my phone but think they were both required

Reply:
You're right, because `.build` can "create" a query, even if it does not yet exist in the cache.

Are there methods in tRPC that could kick off a `tanstack/query` query without going through the cache manager? Like if we use `useQuery`, it immediately creates the query.

- if there are such methods, then `.build()` seems indeed appropriate
- if there aren't, maybe `const query = queryCache.get(queryHash)` could suffice

Also, maybe it doesn't matter and TkDodo can tell us more!